internal/sql/log.go

```diff
@@ -0,0 +1,159 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// The following types are wrappers for the sql package types, adding logging
```

Comment:
That looks cool indeed. I'll add it to the todo list of things to change as we get rid of gorm but for now I'd roll with this since it's "free" since we got it from `hostd` and pretty close to what we have right now.

I'd like to spend a dedicated PR on improving the logging and spend some time testing an external lib before adapting it but for that we need to hook this new code up to some of our slow queries.